### PR TITLE
add guardian module that hooks into guardian db

### PR DESCRIPTION
**Pull request by bbrewer-forge**
https://github.com/SmartColumbusOS/scosopedia/issues/218

### CODE REVIEW

File: apps/discovery_api/lib/discovery_api_web/auth/error_handler.ex [hunk 1]
```diff
@@ -7,6 +7,11 @@ defmodule DiscoveryApi.Auth.ErrorHandler do
   @impl Guardian.Plug.ErrorHandler
   def auth_error(conn, error, _opts) do
     Logger.error("Auth failed: #{inspect(error)}")
-    DiscoveryApiWeb.RenderError.render_error(conn, 401, "Unauthorized")
+
+    error_message = "Unauthorized"
```
Comment (bbrewer-forge) [inline apps/discovery_api/lib/discovery_api_web/auth/error_handler.ex:11]: This was actually rendering early, before the retry in verify auth0 header could hit

File: apps/discovery_api/test/integration/discovery_api/auth/token_handler_test.exs [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      options = []
+
+      assert {:error, _} = TokenHandler.on_verify(
+        new_claims_without_aud,
+        token,
+        options
+      )
+    end
+
+    # given a revoked token, but missing/not missing yada yada yada
```
Comment (bbrewer-forge) [inline apps/discovery_api/test/integration/discovery_api/auth/token_handler_test.exs:94]: ewps

File: apps/discovery_api/test/integration/discovery_api/auth/auth_test.exs [hunk 3]
```diff
@@ -7,13 +7,14 @@ defmodule DiscoveryApi.Auth.AuthTest do
   import SmartCity.TestHelper, only: [eventually: 3]
 
   alias DiscoveryApi.Auth.GuardianConfigurator
+  alias DiscoveryApiWeb.Auth.TokenHandler
   alias DiscoveryApi.Test.Helper
   alias DiscoveryApi.Test.AuthHelper
   alias DiscoveryApi.Schemas.Users
   alias DiscoveryApi.Schemas.Visualizations
   alias DiscoveryApi.Repo
+  require IEx
```
Comment (bbrewer-forge) [inline apps/discovery_api/test/integration/discovery_api/auth/auth_test.exs:16]: ewps

File: apps/discovery_api/test/integration/discovery_api/auth/auth_test.exs [hunk 4]
```diff
@@ -65,7 +66,7 @@ defmodule DiscoveryApi.Auth.AuthTest do
       auth0_setup()
       |> on_exit()
 
-      user = Helper.create_persisted_user(AuthHelper.valid_jwt_sub())
+      {user, _} = AuthHelper.login()
```
Comment (bbrewer-forge) [inline apps/discovery_api/test/integration/discovery_api/auth/auth_test.exs:69]: Not sure I love this helper, as it's a bit ambiguous what token belongs to it. I may just have it return the token it used. Eh

File: apps/discovery_api/lib/discovery_api_web/auth/token_handler.ex [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    case Guardian.DB.after_encode_and_sign(resource, @token_type, storable_claims, token) do
+      {:ok, _} -> {:ok, token}
+      e -> e
+    end
+  rescue
+    e -> {:error, e}
+  end
+
+  def on_verify(claims, token, options) do
+    case Keyword.get(options, :store_token, false) do
```
Comment (bbrewer-forge) [inline apps/discovery_api/lib/discovery_api_web/auth/token_handler.ex:40]: Not particularly proud of any of this...

File: apps/discovery_api/test/integration/discovery_api/auth/auth_test.exs [hunk 6]
```diff
@@ -170,22 +183,140 @@ defmodule DiscoveryApi.Auth.AuthTest do
...
+      |> HTTPoison.post!(
+        "",
+        Authorization: "Bearer " <> token
+      )
+
+      assert {_, 401} = AuthHelper.login(subject, token)
+    end
+
+
+    test "logout is not iDEMPoTent" do
```
Comment (JarredOlson) [inline apps/discovery_api/test/integration/discovery_api/auth/auth_test.exs:218]:
```suggestion
    test "logout is not idempotent" do
```

File: apps/discovery_api/test/integration/discovery_api/auth/token_handler_test.exs [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+        token,
+        options
+      )
+    end
+  end
+
+  defp claims(overrides \\ %{}) do
+    demo_issuer = "https://smartcolumbusos-demo.auth0.com/"
+    demo_client_id = "sfe5fZzFXsv5gIRXz8V3zkR7iaZBMvL0"
+    darsh_subject = "auth0|5e3066daf048aa0e71bdd77e"
```
Comment (JarredOlson) [inline apps/discovery_api/test/integration/discovery_api/auth/token_handler_test.exs:218]: are any of these legit?  If so, and they do not need to be please obfuscate.

Comment (bbrewer-forge) [reply apps/discovery_api/test/integration/discovery_api/auth/token_handler_test.exs:218]: Yeah, and they can be masked. Everything but the keyid, honestly

**Pull request merged**